### PR TITLE
feat: report filter by id

### LINKED ISSUE #126: Filter by `id` in report
**Is your feature request related to a problem? Please describe.**
Currently, we have a filtering mechanism for severity and tags. We should have an option to filter out by query `id`,

**Describe the solution you'd like**
A flag of `--id` or `--query-id` to specify a query ID such as: `--query-id RQ-1`. We should be able to specify one or multiple query IDs.

**Describe alternatives you've considered**
-

**Additional context**
-

### PR DESCRIPTION
**Pull request by tal66**
closes #126

### CODE REVIEW

File: src/cmdline.py [hunk 1]
```diff
@@ -251,3 +260,16 @@ def raven() -> None:
             COMMAND_FUNCTIONS[args.command]()
     else:
         parser.print_help()
+
+
+def validate_query_ids(ids_arg) -> list:
```
Comment (elad-pticha) [inline src/cmdline.py:265]: I think this part should move to `common` as a util function.

**Review by elad-pticha — CHANGES_REQUESTED** [review]
LGTM! Please also update the `README.md` with the the new help menu of the reporter.

**Comment by tal66** [comment]
done. 
- review: moved the validate function to `utils`
- readme: copied from `-h`

**Comment by elad-pticha** [comment]
Please rebase from the main branch 🙏

File: src/common/utils.py [hunk 2]
```diff
@@ -121,3 +122,17 @@ def is_url_contains_a_token(url) -> bool:
 
 def str_to_bool(s: str) -> bool:
     return bool(int(s))
+
+
+def validate_query_ids(ids_arg) -> list:
```
Comment (elad-pticha) [inline src/common/utils.py:127]: Can you please add typing for `ids_arg`?

**Comment by tal66** [comment]
did the rebase (and squash) and the last request

**Comment by elad-pticha** [comment]
Hey @tal66,
Thank you for the changes! A quick update: we are working on a CLA agreement before merging this PR.
We will make sure to keep you updated when we are ready.

**Comment by elad-pticha** [comment]
Hey @tal66,
We created a CLA for new contributors!
Please take a look at our [Contributing Guidelines](https://github.com/CycodeLabs/raven/wiki/Contributing-Guidelines#cla-contributor-license-agreement) for more information.

**Comment by tal66** [comment]
strict :)
sent it just now

**Comment by elad-pticha** [comment]
Great!
Feel free to rebase and fix issues if there's any 🙏

**Review by elad-pticha — APPROVED** [review]
LFTM

**Comment by elad-pticha** [comment]
Due to branch protection rules, we require all commits to be signed. Can you please make sure to sign the commit?

**Comment by tal66** [comment]
oops PR got closed 'cause i renamed the branch, due to a failing test (branch_prefix="^(feat|fix|refactor|chore|docs|test|library)\/[a-z0-9._-]+$"). maybe you can reopen it

**Comment by elad-pticha** [comment]
For now, it is ok that the convention check is failing. However, It is important the commit will be signed.

**Comment by elad-pticha** [comment]
Signed commits (using the GPG key) should look like this (contains the verified icon):

image

You can read more [here](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) about commit signing.

**Comment by tal66** [comment]
ok, created a temp key and now the commit appears to be verified (other repos don't require this)

File: src/config/config.py [hunk 3]
```diff
@@ -58,6 +58,8 @@
     "endoflife",
     "reconnaissance",
 ]
+LAST_QUERY_ID = 16
```
Comment (oreenlivnicode) [inline src/config/config.py:61]: I'm concerned that when someone adds a new query to the query library, they might forget to update the constant. As I see it, we have two options: make this constant a dynamic variable that reflects the number of query files in the query directory, or introduce a test to ensure the number of queries in the directory matches the constant. 
What are your thoughts?

**Review by oreenlivnicode — CHANGES_REQUESTED** [review]
Looks great :)
I have requested one change

**Review by oreenlivnicode — APPROVED** [review]
Great!

**Pull request merged**